Sources/BraintreeUIComponents/PaymentButtonStyle.swift

```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+
+/// Generic button style for payment buttons that handles interaction states
+struct PaymentButtonStyle<Color: PaymentButtonColorProtocol>: ButtonStyle {
+    
+    let color: Color
+    let width: CGFloat?
+    
+    /// This is the width range for the payment button.
+    private var widthRange: CGFloat {
+        guard let width else { return 300 }
+        return min(max(width, 131), 300)
+    }
+    
+    func makeBody(configuration: Configuration) -> some View {
+        HStack {
+            if let logoImageName = color.logoImageName {
+                Image(logoImageName, bundle: .uiComponents)
+                    .resizable()
+                    .scaledToFit()
+                    .frame(height: 14)
+            }
+        }
+        .frame(width: widthRange)
+        .frame(height: 45)
+        .background(configuration.isPressed ? color.tappedButtonColor : color.backgroundColor)
+        .cornerRadius(4)
+        .overlay(
+            RoundedRectangle(cornerRadius: 4)
+                .stroke(
+                    color.hasOutline ? .black : .clear,
+                    lineWidth: color.hasOutline ? 1 : 0
```

Comment:
Are we able to double check the figma for this to make sure 1px is correct or should we be using like `0.75` or something smaller? It looks a little big currently but my eyes may be playing tricks.